metricflow/sql_clients/base_sql_client_implementation.py

```diff
@@ -203,3 +217,61 @@ def close(self) -> None:  # noqa: D
     def render_execution_param_key(self, execution_param_key: str) -> str:
         """Wrap execution parameter key with syntax accepted by engine."""
         return f":{execution_param_key}"
+
+    def async_query(  # noqa: D
+        self,
+        statement: str,
+        bind_parameters: SqlBindParameters = SqlBindParameters(),
+        tags: SqlRequestTagSet = SqlRequestTagSet(),
+    ) -> SqlRequestId:
+        with self._state_lock:
+            request_id = SqlRequestId(f"mf_rid__{random_id()}")
+            thread = SqlRequestExecutorThread(
+                sql_client=self,
+                request_id=request_id,
+                statement=statement,
+                bind_parameters=bind_parameters,
+                tags=tags.add_request_id(request_id),
```

Comment:
You're doing this twice. Is that intentional? It's a bunch of objects to create and throw away.

I think I like it better if request_id is special and internal - always added, etc. as it is in the SqlRequestExecutorThread - and this `tags` parameter is really for arbitrary tags that people want to pass through.

Reply:
Moved to `SqlRequestExecutorThread`.